src/clj/qbits/jet/client/http.clj

```diff
@@ -369,13 +385,23 @@
                             (.close provider)
                             provider)))
 
-    (when body
+    (if (and body (= (.asString HttpVersion/HTTP_2) version))
+      ;; HTTP/2 requests with a body need the trailer to set be set as late as possible
```

Comment:
This comment should dive a bit deeper on the reasoning.

Reply:
Added motivation for avoiding sending empty trailer.